python/cuml/cuml/ensemble/randomforest_common.pyx

```diff
@@ -534,3 +644,80 @@ class BaseRandomForestModel(Base, InteropMixin):
                 align_bytes=align_bytes,
             )
         return fil_model
+
+    @abstractmethod
+    def _compute_oob_score_metric(self, y_true, oob_predictions, valid_mask):
+        """
+        Compute the OOB score metric for the specific model type.
+
+        Parameters
+        ----------
+        y_true : array-like
+            True labels
+        oob_predictions : array-like
+            OOB predictions (1D for regression, 2D for classification)
+        valid_mask : array-like
+            Boolean mask indicating which samples have OOB predictions
+
+        Returns
+        -------
+        float
+            The computed score
+        """
+        pass
+
+    def _compute_oob_score(self, X, y):
+        """
+        Compute OOB score using per-tree predictions and bootstrap masks.
+        """
+        # Get per-tree predictions using FIL
+        fil_model = self.as_fil()
+        per_tree_preds = fil_model.predict_per_tree(X)
+        # Shape: (n_samples, n_trees) for regression
+        #        (n_samples, n_trees, n_classes) for classification
```

Comment:
Is this shape of `per_tree_preds`? If so, can you move the comment to before the line and maybe clarify as "per_tree_threads shape" or something?